### DISCUSSION TITLE
Wix Burn - programatically change the UpdateLocation - maybe in OnDetectUpdateBegin?

### DISCUSSION THREAD

**Discussion by woscostametaxa** [wixtoolset/issues#7174] [Questions] [answered]
I want to build a Web-Setup with auto-update capabilities; so far I succeeded in a local 

In the OnDetectUpdateBegin event I can set the ea.Result to Result.Ok if I want to check for upates - but I would like to be able to change the UpdateLocation during runtime, similar to what the Visual Studio Installer allows - to have different ReleaseFeeds, something like PROD or TESTING or PREVIEW.

Is there any possiblity to tell the Engine to set the UpdateLocation to a different URI? It would seem that the OnDetectUpdateBegin event would be perfect to change that, but the ea.UpdateLocation only has a Get, and I did not find a Engine.SetUpdateLocation or anything similar.

In our context of industrial software it could be important for customers to not add all released patches as updates for machines, but rather sometimes "pick and choose" basically which feed entries to add to a custom feed of a customer. Currently the Burn BA is in WiX 3.11, but from reading the changelog there haven' been changes in this regard in the 4.x?

If I am not mistaken, the VStudio Installer is based upon Wix Burn (?) why I would presume, somewhere, somehow it should be possible to change the URI.

I attach the link where I asked similarly in [Stackoverflow](https://stackoverflow.com/questions/74783298/wix-burn-ba-download-payloads-behind-oauth-and-is-there-a-possibility-to-chan):

**Comment by barnson** [comment, marked as answer]
WiX v4 adds IBootstrapperEngine.SetUpdateSource to support this.
